src/main/java/Cars.java

```diff
@@ -0,0 +1,29 @@
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Cars {
+    private final List<Car> cars;
+    private GameCondition gameCondition;
+
+    public Cars(List<Car> cars){
+        this.cars = cars;
+    }
+
+    public List<Car> getCars() {
+        return cars;
+    }
+
+    public void setMoveStrategy(GameCondition gameCondition){
+        this.gameCondition = gameCondition;
+    }
+
+    public List<Integer> getCarRecords() {
+        return cars.stream()
+                .map(Car::getNumberOfAdvance)
+                .collect(Collectors.toList());
+    }
+
+    public void advanceCars() {
+        cars.forEach(car -> car.advance(gameCondition));
+    }
```

Comment:
```suggestion
    public void advanceCars(GameCondition gameCondition) {
        cars.forEach(car -> car.advance(gameCondition));
    }
```
이렇게 메소드를 실행할 때 전략을 받아서 실행하면 더 유연한 코드가 되지 않을까요!?